features/netsocket/UDPSocket.cpp

```diff
@@ -33,6 +33,17 @@ nsapi_protocol_t UDPSocket::get_proto()
     return NSAPI_UDP;
 }
 
+int UDPSocket::join_multicast_group(const SocketAddress &address)
```

Comment:
Is this needed in addition to the socket option? Having both feels kind of the worst of both worlds - the socket option switch statement has the disadvantage (in an embedded system) that it pulls in all the stack code at link time, even for cases that are never invoked. This is a neat separate API that would allow linker exclusion, but then goes through the switch anyway.

Don't feel terribly strongly, maybe you should have the leave function to match if you have this.